File: backend/src/dto/alert_dto.py
from typing_extensions import TypedDict, NotRequired

class AlertDto(TypedDict):
    id:NotRequired[str]
    title: NotRequired[str]
    category:str
    subCategory:str
    origin:str
    assignedTo:str
    status:str
    creationTime:NotRequired[str]
    conclusionTime:NotRequired[str]
    description:NotRequired[str]
    resolution:NotRequired[list[str]]
    priority:int

File: frontend-app/src/domain/AlertDTO.ts
interface AlertDTO {
    title: string;
    category: string;
    subCategory: string;
    origin: string;
    assignedTo: string;
    status: string;
    creationTime: string;
    conclusionTime: string;
    description: string;
    resolution: string[];
}
export default AlertDTO;